lib/Extractor/Solver.cpp

```diff
@@ -281,16 +285,17 @@ class MemCachingSolver : public Solver {
 
   std::error_code infer(const BlockPCs &BPCs,
                         const std::vector<InstMapping> &PCs,
-                        Inst *LHS, Inst *&RHS, InstContext &IC) override {
+                        Inst *LHS, std::vector<Inst *> &RHSs,
+                        InstContext &IC) override {
     ReplacementContext Context;
     std::string Repl = GetReplacementLHSString(BPCs, PCs, LHS, Context);
     const auto &ent = InferCache.find(Repl);
     if (ent == InferCache.end()) {
       ++MemMissesInfer;
-      std::error_code EC = UnderlyingSolver->infer(BPCs, PCs, LHS, RHS, IC);
+      std::error_code EC = UnderlyingSolver->infer(BPCs, PCs, LHS, RHSs, IC);
       std::string RHSStr;
-      if (!EC && RHS) {
-        RHSStr = GetReplacementRHSString(RHS, Context);
+      if (!EC && !RHSs.empty()) {
+        RHSStr = GetReplacementRHSString(RHSs.front(), Context);
```

Comment:
seems weird to take the first element here, is this a placeholder for something different?

Reply:
This PR does not support cache dumping yet, so only print the first RHS in the list. I'll add support for cache dumping soon.

Reply:
I'd say don't do the caching part of this yet, until we're sure we want it

Reply:
Agreed.

Reply:
Sounds good, should be possible to prototype without caching.